src/main/java/ru/spbstu/repository/VariantRepository.java

```diff
@@ -0,0 +1,27 @@
+package ru.spbstu.repository;
+
+import org.springframework.data.jpa.repository.JpaSpecificationExecutor;
+import org.springframework.data.repository.CrudRepository;
+import org.springframework.stereotype.Repository;
+import ru.spbstu.model.Variant;
+
+import javax.persistence.criteria.CriteriaBuilder;
+import javax.persistence.criteria.Predicate;
+import javax.persistence.criteria.Root;
+import java.util.Optional;
+
+@Repository
+public interface VariantRepository extends CrudRepository<Variant, Long>, JpaSpecificationExecutor<Variant> {
+  default Optional<Variant> findVariant(String chromosome, Long position, String referenceBase, String alternateBase) {
```

Comment:
The current implementation is ok.
But for the simple CRUD functionality we can use https://www.baeldung.com/spring-data-derived-queries

In this case we can just add abstract method with name like:
`findByChromosomeAndPositionAndReferenceBaseAndAlternativeBase`
It's a little bit hard to read but in this case we delegating the implementation to the hibernate.

Reply:
to be honest, i don't like the method with such long name.

Reply:
@RomanEgel, I agree. Then jpql can be used (in @Query annotation).

Reply:
actually i don't understand what the problem with criteria api. sometimes out-of-box solutions doesn't fill your needs. 
Jpql is almost sql and since we allow nullable ref and alt fields we should take care of null situations which doesn't look pretty in sql.